gc/base/TLHAllocationInterface.cpp

```diff
@@ -282,6 +282,11 @@ MM_TLHAllocationInterface::flushCache(MM_EnvironmentBase *env)
 	MM_GCExtensionsBase *extensions = env->getExtensions();
 	
 #if defined(OMR_GC_THREAD_LOCAL_HEAP)	
+	/* update traceAllocationBytes with allocatedSizeInsideTLH during flushing TLH Cache */
+	uintptr_t allocatedSizeInsideTLH = env->getAllocatedSizeInsideTLH();
+	env->_oolTraceAllocationBytes += allocatedSizeInsideTLH;
+	env->_traceAllocationBytes += allocatedSizeInsideTLH;
```

Comment:
Hm, should not this be _owningEnv. Caller's is probably a thread triggered GC/acquired Exclusive VM access, and iterates through all threads and flush cache on their behalf. 

If test is just single threaded, this may help, but I suspect is not good enough with multiple Mutators.

Reply:
updated with _owningEnv.

Reply:
Unrelated to this change, but I'm wondering if there is a problem with rest of the code in this method - shouldn't we be merging _owningEnv's stats etc!?

Reply:
Actually, seems like at the call site the argument passed was not the caller, but target thread. Hence env == _owningEnv, and even your original change was ok.
This is not really how things should be done, but further changes would be out of scope of this fix. Let's keep this simple.

```
GC_OMRVMInterface::flushCachesForGC(MM_EnvironmentBase *env)
{
...
	while((omrVMThread = threadListIterator.nextOMRVMThread()) != NULL) {
...
		MM_EnvironmentBase* threadEnv = MM_EnvironmentBase::getEnvironment(omrVMThread);
...
		GC_OMRVMThreadInterface::flushCachesForGC(threadEnv);  <<<< this should have passed 2 params: calling Env, and target Env

------------
GC_OMRVMThreadInterface::flushCachesForWalk(MM_EnvironmentBase *env)
{
	env->_objectAllocationInterface->flushCache(env);  << this should be: targetEnv->_objectAllocationInterface->flushCache(callerEnv)

-----
void
MM_TLHAllocationInterface::flushCache(MM_EnvironmentBase *env)
//// should deal with _owningEnv all the time

```

Reply:
@amicic I made an issue for this https://github.com/eclipse/omr/issues/5345